File: movie-client/src/main/java/top/tonydon/util/observer/ClientObserver.java
package top.tonydon.util.observer;


import top.tonydon.message.common.BindMessage;
import top.tonydon.message.common.BulletScreenMessage;
import top.tonydon.message.common.MovieMessage;
import top.tonydon.message.server.ServerConnectMessage;
import top.tonydon.message.server.ServerResponseMessage;

public interface ClientObserver {
    void onConnected(ServerConnectMessage message);

    void onMovie(MovieMessage message);

    void onBind(BindMessage message);

    void onUnbind();

    void onOffline();

    void onBulletScreen(BulletScreenMessage message);

    void onServerMessage(ServerResponseMessage responseMessage);
}

File: client/src/main/java/top/tonydon/syncplayer/util/observer/ClientObserver.kt
package top.tonydon.syncplayer.util.observer

import top.tonydon.syncplayer.message.common.MovieMessage

interface ClientObserver {
    fun onAction(code: Int)
    fun onString(code: Int, content: String)
    fun onMovie(message: MovieMessage)
    fun onError(exception: Exception)
}
